ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatRecord.java

```diff
@@ -869,100 +874,247 @@ public static BlobProperties deserializeBlobPropertiesRecord(CrcInputStream crcS
+      return new UpdateRecord(accountId, containerId, udpateTimeInMs, new DeleteRecord());
+    }
+  }
+
+  /**
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -- - - - - - -
+   * |         |               |               |               |             |            |
+   * | version |   AccountId   |  ContainerId  |  UpdateTime   |  Sub record |    Crc     |
+   * |(2 bytes)|    (2 byte2)  |   (2 bytes)   |   (8 bytes)   |  (n bytes)  |  (8 bytes) |
+   * |         |               |               |               |             |            |
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -- - - - - - -
+   *  version       - The version of the update record
+   *
+   *  AccountId     - AccountId that the blob belongs to
+   *
+   *  ContainerId   - ContainerId that the blob belongs to
+   *
+   *  UpdateTime    - Time of update in Ms
+   *
+   *  Sub record    - the sub record related to this update
+   *
+   *  Crc           - The crc of the update record
+   *
+   */
+  public static class Update_Format_V3 {
+
+    private static final int RECORD_TYPE_FIELD_SIZE_IN_BYTES = Short.BYTES;
+    private static final int ACCOUNT_ID_FIELD_SIZE_IN_BYTES = Short.BYTES;
+    private static final int CONTAINER_ID_FIELD_SIZE_IN_BYTES = Short.BYTES;
+    private static final int UPDATE_TIME_FIELD_SIZE_IN_BYTES = Long.BYTES;
+
+    /**
+     * @param type the type of the sub record in this update record.
+     * @return the size of the record if the record were serialized
+     */
+    public static int getRecordSize(UpdateRecord.Type type) {
+      int subRecordSize;
+      switch (type) {
+        case DELETE:
+          subRecordSize = Delete_Format_V1.getRecordSize();
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown update record type: " + type);
+      }
+      return Version_Field_Size_In_Bytes + ACCOUNT_ID_FIELD_SIZE_IN_BYTES + CONTAINER_ID_FIELD_SIZE_IN_BYTES
+          + UPDATE_TIME_FIELD_SIZE_IN_BYTES + RECORD_TYPE_FIELD_SIZE_IN_BYTES + subRecordSize + Crc_Size;
+    }
+
+    /**
+     * Serializes {@code updateRecord} into {@code outputBuffer} in version 3.
+     * @param outputBuffer the buffer to write the serialized bytes into.
+     * @param updateRecord the {@link UpdateRecord} to serialize.
+     */
+    public static void serialize(ByteBuffer outputBuffer, UpdateRecord updateRecord) {
+      int startOffset = outputBuffer.position();
+      outputBuffer.putShort(Update_Version_V3);
+      outputBuffer.putShort(updateRecord.getAccountId());
+      outputBuffer.putShort(updateRecord.getContainerId());
+      outputBuffer.putLong(updateRecord.getUpdateTimeInMs());
+      outputBuffer.putShort((short) updateRecord.getType().ordinal());
+      switch (updateRecord.getType()) {
+        case DELETE:
+          Delete_Format_V1.serialize(outputBuffer, updateRecord.getDeleteRecord());
```

Comment:
Call this Delete_SubFormat_V1 or something? I was initially confused that you were reusing existing Delete_Format_V1.

Reply:
Will chat about naming with you before making changes

Reply:
Renamed